app/scripts/app.js

```diff
@@ -58,10 +58,15 @@ angular.module('yourStlCourts').config(function($stateProvider, $urlRouterProvid
       resolve: {
         court: function ($stateParams,Courts) {
           if ($stateParams.courtId != ""){
-            return Courts.findById($stateParams.courtId)
+          return (Courts.findById($stateParams.courtId)
+            .then(function(court){ //a court was found
+              return court;},
+            function(err){ //something about the find Court by Court Id went wrong
+              return null;
```

Comment:
I don't think this is what we want....we don't want them to go to the page at all. null court doesn't give them any value to proceed to that page. Typically, you'll want to toast an error message here. I'm not sure if the ui-router will stop the transition on a failed promise, though. Which is fine (for now), but we don't want to purposefully set null when null doesn't give us a valid workflow to the next page. Look for ```$stateChangeError``` on https://github.com/angular-ui/ui-router/wiki. This is not something I'm asking you to setup but it's something we should probably add (by calling ```event.preventDefault()``` ) in the ```stateChangeError``` callback function. Point, is, show the user an error via the toast vs return null.